src/extra/others/fragment/lv_fragment.c

```diff
@@ -0,0 +1,392 @@
+/**
+ * @file lv_fragment.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_fragment.h"
+
+#if LV_USE_FRAGMENT
+
+#include "../../../extra/widgets/msgbox/lv_msgbox.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct manager_stack_t {
+    const lv_fragment_class_t * cls;
+    lv_fragment_t * instance;
+    bool obj_created;
+    bool destroying_obj;
+    bool dialog;
+    struct manager_stack_t * prev;
+} manager_stack_t;
+
+struct _lv_fragment_manager_t {
+    lv_obj_t * container;
+    lv_fragment_t * parent;
+    manager_stack_t * top;
+};
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static manager_stack_t * item_new(const lv_fragment_class_t * cls);
+
+static lv_fragment_t * item_create_fragment(lv_fragment_manager_t * manager, manager_stack_t * item,
+                                            void * args);
+
+static void item_create_obj(lv_fragment_manager_t * manager, manager_stack_t * item, lv_obj_t * parent,
+                            const lv_obj_class_t * check_type);
+
+static void item_destroy_obj(lv_fragment_manager_t * manager, manager_stack_t * item);
+
+static void item_destroy_fragment(manager_stack_t * item);
+
+static void obj_cb_delete(lv_event_t * event);
+
+static void fragment_destroy_obj(lv_fragment_t * fragment);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_fragment_manager_t * lv_fragment_manager_create(lv_obj_t * container, lv_fragment_t * parent)
+{
+    LV_ASSERT(container);
+    lv_fragment_manager_t * instance = lv_mem_alloc(sizeof(lv_fragment_manager_t));
+    instance->parent = parent;
+    instance->container = container;
+    instance->top = NULL;
+    return instance;
```

Comment:
I prefer to let users do by themselves. Because we can still keep fragment instance after view destroyed, so considering we may support keeping nested manager instance, we should leave this to the user.